### PR TITLE
Added Purpose of chunks

### PR DESCRIPTION
**Pull request by RobinAtherton**

### CODE REVIEW

File: LaTeX/varroa-dokumentation.tex [hunk 1]
```diff
@@ -21,6 +21,7 @@
 \input{Parts/Chapter2.tex}
 \input{Parts/Chapter3.tex}
 \input{Parts/Chapter4.tex}
+\input{Parts/Chapter-Chunk-Concept}
```
Comment (AcidSepp) [inline LaTeX/varroa-dokumentation.tex]: please add this to the architecture chapter

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+\chapter{Chunk Concept}
+\section{Definition}
+A chunk is a type of container designed to distribute data from the Commander to its Agents.
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]:
```suggestion
A chunk is a work package of the Scenario execution designed to be distributed from the Commander to its Agents.
```

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+It contains:
+\begin{itemize}
+	\item the amount of clients a chunk represents.
+	\item the commands that are applied to the clients in the chunk.
+	\item information concerning the MQTT-specific properties.
+	\item variables defining lifecycle properties.
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]: too generic, specify the variables and their values

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 4]
```diff
@@ -0,0 +1,20 @@
+\chapter{Chunk Concept}
+\section{Definition}
+A chunk is a type of container designed to distribute data from the Commander to its Agents.
+It contains:
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]: maybe specify the different forms in which chunks can exist

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+\begin{itemize}
+	\item the amount of clients a chunk represents.
+	\item the commands that are applied to the clients in the chunk.
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]:
```suggestion
	\item the commands that are executed by the clients
```

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+	\item information concerning the MQTT-specific properties.
+	\item variables defining lifecycle properties.
+\end{itemize}
+
+\section{Purpose of Chunks}
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]: We do not need to describbe why we use chunks, we need to explain the concept itself.

File: LaTeX/Parts/Chapter-Chunk-Concept.tex (hunk 2 above)
Comment (AcidSepp) [reply LaTeX/Parts/Chapter-Chunk-Concept.tex]: It is then handled by the Agent it is distributed to.
The Agent now creates or uses existing MQTT Agents to execute the Chunk.

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+Due to the distributed nature of Varroa it was necessary to be able to spread the information contained in the scenario.
+To secure a deterministic distribution of a scenario's data the chunks had to be evenly spread across the initialized Agents.
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]: factually incorrect. Even spreading of the chunks does not induce determinism. Determinism means that every chunk is distributed to the exact same Agent in every Varroa Execution.

File: LaTeX/Parts/Chapter-Chunk-Concept.tex [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+To ensure this determinism a mechanism was introduced combining a round-robin selection of chunks with a handshake mechanism of the agents.
```
Comment (AcidSepp) [inline LaTeX/Parts/Chapter-Chunk-Concept.tex]: 👍

**Review by AcidSepp — CHANGES_REQUESTED** [review]

File: LaTeX/Parts/Chapter-Chunk-Concept.tex (hunk 4 above)
Comment (RobinAtherton) [reply LaTeX/Parts/Chapter-Chunk-Concept.tex]: ?

**Review by AcidSepp — APPROVED** [review]

**Pull request merged**